UI/frontend-plugins/decklink-output-ui/DecklinkOutputUI.cpp

```diff
@@ -0,0 +1,67 @@
+#include "DecklinkOutputUI.h"
+#include <obs-module.h>
+#include <util/platform.h>
+#include "decklink-ui-main.h"
+
+DecklinkOutputUI::DecklinkOutputUI(QWidget *parent)
+		: QDialog(parent),
+		  ui(new Ui_Output) {
+	ui->setupUi(this);
+
+	propertiesView = nullptr;
+
+	connect(ui->startOutput, SIGNAL( released() ), this, SLOT(StartOutput()));
+	connect(ui->stopOutput, SIGNAL( released() ), this, SLOT(StopOutput()));
+}
+
+void DecklinkOutputUI::ShowHideDialog()
+{
+	SetupPropertiesView();
+
+	setVisible(!isVisible());
+}
+
+void DecklinkOutputUI::SetupPropertiesView()
+{
+	if (propertiesView)
+		delete propertiesView;
+
+	obs_data_t *settings = obs_data_create();
+
+	OBSData data = load_settings();
+	if (data != nullptr)
+		obs_data_apply(settings, data);
+
+	propertiesView = new OBSPropertiesView(settings,
+						  "decklink_output",
+						  (PropertiesReloadCallback) obs_get_output_properties,
+						  170);
+
+	ui->propertiesLayout->addWidget(propertiesView);
+	obs_data_release(settings);
+}
+
+void DecklinkOutputUI::SaveSettings()
+{
+
+	const char *modulePath = obs_module_get_config_path(obs_current_module(), "");
```

Comment:
Style:  This line is too long and needs to be broken.

Does this need to explicitly be a const char* for `os_mkdirs` below?  It seems like `obs_data_save_json_safe` expects `path` to be a const char*, but you use a char* for that.  If you were able to safely remove the `const`, this line does not need broken.